test/unit-tests/fiber.mlt

```diff
@@ -0,0 +1,163 @@
+open Dune;;
+open Stdune;;
+open Fiber;;
+open Fiber.O;;
+
+let failing_fiber () =
+  Fiber.yield ()
+  >>= fun () ->
+  raise Exit
+;;
+
+let long_running_fiber () =
+  let rec loop n =
+    if n = 0 then
+      Fiber.return ()
+    else
+      Fiber.yield ()
+      >>= fun () ->
+      loop (n - 1)
+  in
+  loop 10
+;;
+
+let ivar : unit Fiber.Ivar.t = Fiber.Ivar.create ();;
+
+let getter_fiber () =
+  Fiber.Ivar.read ivar
+;;
+
+[%%expect{|
+val failing_fiber : unit -> 'a t = <fun>
+val long_running_fiber : unit -> unit t = <fun>
+val ivar : unit Ivar.t = <abstr>
+val getter_fiber : unit -> unit t = <fun>
+|}]
+
+Fiber.run (Fiber.collect_errors failing_fiber)
+[%%expect{|
+- : ('a, exn list) Stdune.result = Error [Exit]
+|}]
+
+let success = ref false;;
+
+try
+  Fiber.run (Fiber.collect_errors getter_fiber);
+  ()
+with Fiber.Never ->
+  success := true
```

Comment:
No need to go through a reference in this case, we can return a boolean directly